mobile/src/employee-details/employee-details.tsx

```diff
@@ -119,18 +147,34 @@ export class EmployeeDetailsImpl extends Component<EmployeeDetailsProps & Employ
             return calendarEvents !== nextCalendarEvents;
         }
 
+        const requests = this.props.requests;
+        const nextRequests = nextProps.requests;
+
+        if (nextRequests && !nextRequests.equals(requests)) {
+            return true;
+        }
+
+        const approvals = this.props.approvals;
+        const nextApprovals = nextProps.approvals;
+
+        if (!approvals.equals(nextApprovals)) {
+            return true;
+        }
+
         return false;
     }
 
+    //----------------------------------------------------------------------------
     public componentDidMount() {
         this.props.loadCalendarEvents(this.props.employee.employeeId);
         this.props.loadUserEmployeePermissions(this.props.employee.employeeId);
     }
 
+    //----------------------------------------------------------------------------
     public render() {
-        const { employee, department, userEmployeePermissions } = this.props;
+        const { employee, department, userEmployeePermissions, userId } = this.props;
```

Comment:
do you still need department in its props?

Reply:
Yes.
```JSX
<TouchableOpacity onPress={this.openDepartment}>
    <StyledText style={contentStyles.department}>
        {this.uppercase(department.abbreviation)}
    </StyledText>
</TouchableOpacity>
```

Reply:
why remove !department then?

Reply:
Because it is not optional %)